docs/introduction/en/storage.md

```diff
@@ -0,0 +1,53 @@
+
+### Solidity memory management mechanism
+
+According to the [Description](https://solidity-cn.readthedocs.io/zh/develop/introduction-to-smart-contracts.html#index-10) of the Solidity document on its memory management and Conflux [Storage Deposit Mechanism](https://mp.weixin.qq.com/s/KKU3XiJqr_Uz5RdlUyQmtA), we can find that the contract storage needs to be maintained with key and value. In general: the length of key is 256bits, and the length of value is also 256bits. The storage space of a smart contract is organized according to the following table, where {0,1}<sup>256</sup> represents a 256-bit string (there are only two values ​​of 0 or 1 in the bit string), **each key/value pair can be understood as an entry**:
+
+Entry | Key/Address({0,1}<sup>256</sup>)   | value({0,1}<sup>256</sup>)  |
+|  :----:  |  :----:  | :----:  |
+1| 0...00000  | 0 |
+2| 0...00001  | 1 |
+3| 0...00002 | 2 |
+...| ... | ... |
+2<sup>256</sup>| f...fffff| 0 |
+
+Since 256bits=32bytes, the corresponding length of two 256bits is: 32+32=64 (Bytes)
+
+Summary of common variables and their corresponding entries in Solidity:
+
+|   Variable   |                            Length                             |                      Definition Method                       |
+| :------: | :---------------------------------------------------------: | :-------------------------------------------------: |
+| Ordinary variables |                   1 ordinary variable corresponds to 1 entry                    |                uint public count=0;                 |
+| mapping  |              Each key of mapping corresponds to 1 entry              |      mapping(address => uint) public balances;      |
+|  array   | Each element of the array corresponds to 1 item, and the length of the array arr.length is an additional item | uint[5] fixedArr = [1,2,3,4,5]; string productname; |
+|  struct  |              Accumulation of the number of entries corresponding to each field in the struct              |  struct Person {uint age;uint stuID;string name;}   |
+
+
+## How to use storage mortgage
+
+When users send a Conflux transaction, they need to fill in a `storageLimit` field (in bytes). The upper limit stipulates that the deposit increase of the deposit payer before and after the transaction is executed shall not exceed the `storage upper limit` multiplied by 1/1024 CFX. If this value is filled in too low, it will cause the deposit to exceed the upper limit after execution and the execution will fail. Therefore, in general, this field needs to be set to a value larger than the actual usage, and the excess part will not generate storage mortgage. However, it is not recommended to fill in too high, because it may cause the sender's balance to be insufficient to pay the deposit, which may result in transaction failure. Fullnode provides the RPC method `cfx_estimateGasAndCollateral` to estimate the storage size that a transaction will use.
+
```

Comment:
Maybe add: "Storage limit plays the same role for storage as gas limit does for execution."